src/app/qgsapplayertreeviewmenuprovider.cpp

```diff
@@ -846,12 +846,6 @@ QMenu *QgsAppLayerTreeViewMenuProvider::createContextMenu()
             QString filterExp = layer->renderer() ? layer->renderer()->legendKeyToExpression( ruleKey, layer, ok ) : QString();
             if ( ok )
             {
```

Comment:
Are there valid cases where `ok` will be false? Because the users would still expect an attribute table then.

If so, it would be probably better to try to get the `filterExp` outside the lambda and disable the Action if it fails.